compiler/shared/main/scala/mlscript/compiler/optimizer/TailRecOpt.scala

```diff
@@ -340,6 +342,9 @@ class TailRecOpt(fnUid: FreshInt, classUid: FreshInt, tag: FreshInt, raise: Diag
                         // If this assignment overwrites the mod cons value, forget it
                         if containingCtors.contains(assignee) then invalidateAndCont(body)
                         else searchOptCalls(body)
+      case LetMethodCall(names, cls, method, args, body) =>
+        // TODO
+        ???
```

Comment:
Unresolved method calls such as virtual calls cannot be analyzed, so we ignore them
```suggestion
        // method call is unresolved, just ignore it
        searchOptCalls(body)(newMap, src, scc, start, calledDefn, letCallNode, letCtorNode, containingCtors - names.toSet) // `containingCtors - names.toSet` takes care of variable shadowing
```

Reply:
What's the definition of `newMap`?

Reply:
my bad, typo